packages/core/src/odata/common/entity-deserializer.ts

```diff
@@ -195,11 +196,15 @@ export function entityDeserializer(
   >(json: MapType<any>, complexType: ComplexTypeNamespaceT): any {
     return complexType._propertyMetadata
       .map(property => ({
-        ...(typeof json[property.originalName] !== 'undefined' && {
-          [property.name]: isComplexTypeNameSpace(property.type)
-            ? deserializeComplexType(json[property.originalName], property.type)
-            : edmToTs(json[property.originalName], property.type)
-        })
+        ...(typeof json[property.originalName] !== 'undefined' &&
+          json[property.originalName] !== null && {
+            [property.name]: isComplexTypeNameSpace(property.type)
+              ? deserializeComplexType(
```

Comment:
I find this really hard to read. Relating the spread (...) with the tenary operator in the third boolean check is in my opinion to much. I would prefer a seperate function returning the object with a proper return and the condition separated.

Reply:
have you seen this @jjtang1985

Reply:
Actually, initially I had the same feeling. Once I read it carefully, I felt the readability is not bad. So I don't have strong opinion about it. Fortunately, I don't touch this code block. So I'll give this opportunity to the next person;)